geomesa-hbase/geomesa-hbase-datastore/src/main/scala/org/locationtech/geomesa/hbase/data/HBaseQueryPlan.scala

```diff
@@ -83,18 +85,22 @@ case class CoprocessorPlan(filter: HBaseFilterStrategyType,
     val hbaseTable = ds.connection.getTable(table)
 
     import org.locationtech.geomesa.hbase.coprocessor._
-    val byteArray = serializeOptions(coprocessor.configureScanAndFilter(scan, filterList))
+    ds.applySecurity(scan)
+    val byteArray = serializeOptions(coprocessorConfig.configureScanAndFilter(scan, filterList))
 
     val result = GeoMesaCoprocessor.execute(hbaseTable, byteArray)
-
-    coprocessor.reduce(result.toIterator.map(r => coprocessor.bytesToFeatures(r.toByteArray)))
+    val results = result.toIterator.filter(_.size() != 0).map(r => coprocessorConfig.bytesToFeatures(r.toByteArray))
+    coprocessorConfig.reduce(results)
   }
 
-  def calculateScanAndFilterList(ranges: Seq[Scan],
+  def calculateScanAndFilterList(ranges: Seq[Query],
                                  remoteFilters: Seq[(Int, HFilter)]): (Scan, FilterList) = {
     val rowRanges = Lists.newArrayList[RowRange]()
-    ranges.foreach { r =>
-      rowRanges.add(new RowRange(r.getStartRow, true, r.getStopRow, false))
+    ranges.foreach {
+      case g: Get =>
+        rowRanges.add(new RowRange(g.getRow, true, IndexAdapter.rowFollowingRow(g.getRow), false))
```

Comment:
good catch - the `gets` are processed out but that doesn't happen in the coprocessor path